### PR TITLE
[docs] Remove README.md files from docs.opentitan.org

### PR DESCRIPTION
**Pull request by moidx**
This PR removes README.md files from docs.opentitan.org. Until now, we have not been using consistent Hugo markup on README files, causing indexing issues in the website. On the other hand, GitHub doesn't render Hugo markup, making READMEs less ergonomic in repo views. 

Some README files that were referenced in hw documentation were moved to Hugo documentation folders `doc/index.md` or `doc/_index.md`. `doc/index.md` is used to define a leaf webpage, `doc/_index.md` instructs Hugo to render documents in the same folder or subfolders.

This PR is part of issue #13345

### CODE REVIEW

**Review by msfschaffner — APPROVED** [review]
Thanks for the cleanup! LGTM

**Comment by mundaym** [comment]
This PR seems to have broken the https://docs.opentitan.org/sw/device/examples/ page. It might just be that we want to delete that page?

I approve of the general proposal though (avoiding rendering README.md files with Hugo and linking to GitHub rendered versions of the file where appropriate).

**Review by mundaym — APPROVED** [review]

**Comment by moidx** [comment]
> This PR seems to have broken the https://docs.opentitan.org/sw/device/examples/ page. It might just be that we want to delete that page?
> 

Removed the page as it doesn't have any content.

File: hw/dv/sv/alert_esc_agent/doc/index.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-# ALERT_ESC Agent
+---
```
Comment (sriyerg) [inline hw/dv/sv/alert_esc_agent/doc/index.md:1]: In general, is it mandatory to always create a separate `doc` directory? I wonder if `index.md` can live alongside the other sources.

**Pull request merged**